src/client/address_book/impl.h

```diff
@@ -63,6 +63,89 @@ class AddressBookSubscriber;
 /// @brief Address book implementation
 class AddressBook : public AddressBookDefaults {
  public:
+  /// @brief Container for Address Book entry
+  class Entry
+  {
+   public:
+    /// @brief No default copy-constructor
+    Entry(Entry& rhs) = delete;
+
+    /// @brief No default move-constructor
+    Entry(Entry&& rhs) = delete;
+
+    /// @brief Create address book entry from a hostname & identity hash
+    /// @param host Kovri hostname
+    /// @param address Kovri identity hash
+    Entry(std::string host, const kovri::core::IdentHash& address)
+        : m_Host(std::move(host)), m_Address(address)
```

Comment:
>Entry(std::string host,
>m_Host(std::move(host))

I see absolutely no reason to not pass const reference here. Why do you want move semantics here.

Reply:
> Item 41: Consider pass by value for copyable parameters that are cheap to move and always copied.

from [Effective Modern C++ by Scott Meyers](https://www.amazon.com/dp/1491903996)
https://www.safaribooksonline.com/library/view/effective-modern-c/9781491908419/ch08.html

passed by const reference
```cpp
class Entry {
public:
  Entry(std::string &host) : m_host(host) {}
private:
  std::string m_host;
};
```

will give the following result:

```cpp
Entry e1("hello.world");
// Create temporary std::string object - first allocation 
// Copy from temporary std::string to m_host - second allocation

auto lambda = []() -> std::string { return "hello.world" };
Entry e2(lambda());
// temporary std::string returned from lambda - first allocation somewhere inside lambda 
// copy from temporary std::string to m_host - second allocation

std::string v = "hello.world";
const std::string& cf = v;
Entry e3(cf);
// only one copy
```

where as, passed by value 

```cpp
class Entry {
public:
  Entry(std::string host) : m_host(std::move(host)) {}
private:
  std::string m_host;
};
```

will give the following result:

```cpp
Entry e1("hello.world");
// Create host argument from string literal - only allocation 
// move from host to m_host

auto lambda = []() -> std::string { return "hello.world" };
Entry e2(lambda());
// temporary std::string returned from lambda - allocation somewhere inside lambda
// move from temporary rvalue to host argument 
// move from host to m_host

std::string v = "hello.world";
const std::string& cf = v;
Entry e3(cf);
// copy from cf to host argument - allocation
// move from host to m_host
```

Reply:
> Why do you want move semantics here.

It was an attempt at writing an optimized container, along the lines of @btolfa's examples above.

After reading more about move semantics, I was mistaken about how/when to use it.

In the contexts where I see `Entry` being beneficial, one would construct the entry using temporary variables. It seems too expensive to deal with the mental overhead of remembering to construct an `Entry` only on the last use of the temporary variables.

I'll look again at the contexts where `Entry` will be useful, but I think I'll remove explicit use of move semantics.

Reply:
>passed by const reference
>Entry(std::string &host)

@btolfa **This is not const reference. The example you provided is not even accurate to our case nor is it needed for our case**.

@coneiric **KISS**